pkg/api/auth_middleware.go

```diff
@@ -56,8 +61,18 @@ func AuthMiddleware(logger logging.Logger, swagger *openapi3.Swagger, authentica
 			}
 			securityRequirements, err := extractSecurityRequirements(router, r)
 			if err != nil {
-				writeError(w, r, http.StatusBadRequest, err)
-				return
+				// if we can't find the route, assume it's the base one.
+				if _, ok := err.(*routers.RouteError); ok {
+					var defaultErr error
+					securityRequirements, defaultErr = defaultSecurityRequirements(router)
+					if defaultErr != nil {
+						writeError(w, r, http.StatusBadRequest, err)
+						return
+					}
+				} else {
+					writeError(w, r, http.StatusBadRequest, err)
+					return
+				}
```

Comment:
Looks like a nice hack - but you need to extract the `for provider := range securityRequirement {` scope to a function and use it in your `MakeAuthMiddleware` it is check a specific set of security requirements without going throw route information by swagger.
This handler was make to handle `/api/v1` + swagger security requirements which you do not use.

Reply:
But I do use it. I don't want to construct a different set of securityRequirements - I want to use the same ones that exist in our Swagger API. If we add another authentication mechanism to the API, I want it to apply to the gateway as well.

Reply:
In the above we added fallback code that handle route not found that uses /api/v1/users as the route we verify security requirements.
- What will happen when the request route for the gateway will match open api route that doesn't require any security check?
- We need to extract once the security requirements specified in the swagger - not for a specific route